src/rustdoc_gen.rs

```diff
@@ -182,11 +176,39 @@ fn save_placeholder_rustdoc_manifest(
     Ok(placeholder_manifest_path)
 }
 
+#[allow(dead_code)]
+pub(crate) enum CrateType<'a> {
+    Current,
+    Baseline {
+        /// When the baseline is being generated from registry
+        /// and no specific version was chosen, we want to select a version
+        /// that is the same or older than the version of the current crate.
+        highest_allowed_version: Option<&'a semver::Version>,
+    },
+}
+
+pub(crate) struct CrateDataForRustdoc<'a> {
+    pub(crate) crate_type: CrateType<'a>,
+    pub(crate) name: &'a str,
+    // TODO: pass an enum describing which features to enable
+}
+
+impl<'a> ToString for CrateType<'a> {
+    fn to_string(&self) -> String {
+        match self {
+            CrateType::Current => "current",
+            CrateType::Baseline { .. } => "baseline",
+        }
+        .to_string()
+    }
+}
```

Comment:
Let's just make this method separate from the trait. We don't really care about `ToString` as a trait (e.g. we'll never do `impl ToString` or `&dyn ToString` for this type) and can afford to just have a method like `fn type_name(&self) -> &str` instead.
```suggestion
impl<'a> CrateType<'a> {
    fn type_name(&self) -> &'static str {
        match self {
            CrateType::Current => "current",
            CrateType::Baseline { .. } => "baseline",
        }
    }
}
```